rclpy/rclpy/action/server.py

```diff
@@ -405,6 +405,7 @@ async def _execute_get_result_request(self, request_header_and_message):
     async def _execute_expire_goals(self, expired_goals):
         for goal in expired_goals:
             goal_uuid = bytes(goal.goal_id.uuid)
+            self._goal_handles[goal_uuid].destroy()
```

Comment:
I see this works and addresses memory leak.

how about adding dtor in `ServerGoalHandle` to call `destroy` method internally? so that once this object is deleted, it automatically issues dtor to destroy the `goal_handle`?

Reply:
CC: @sloretz

Reply:
Thanks for the comment. I thought so too at the first time, but I fount only a few classes have `__del__` method, so I assumed that the most of the classes in `rclpy` are expected to be called `destroy`. I'm ok to change to call `destroy` in `__del__` but I'd like to ask maintainers' opinion as well.

```bash
$ git grep __del__
rclpy/rclpy/executors.py:    def __del__(self):
rclpy/rclpy/signals.py:    def __del__(self):
rclpy/rclpy/task.py:    def __del__(self):
```

Reply:
this is not hard requirement, so that we can do the refactor later, but lets wait and see other opinion.

CC: @sloretz @clalancette @ivanpauno

Reply:
I see. Thank you for your suggestion. I'll try that!